src/command_line_parser.cc

```diff
@@ -1251,7 +1315,8 @@ TritonParser::Parse(int argc, char** argv)
         break;
         break;
       case OPTION_HTTP_THREAD_COUNT:
-        lparams.http_thread_cnt_ = ParseOption<int>(optarg);
+        lparams.http_thread_cnt_ = ParseOption<int>(
+            optarg, GetOptionName(&long_options[option_index]));
```

Comment:
Instead of passing option name into the `Parse` function just for formatting the exception message. How about `try .. catch`ing the `while(getopt_long(...))` loop and repackage the exception with the option name on exception.

Reply:
The function code definitely stand to be refactored further but I think is a good step.

Reply:
Refactored as suggested